enhancements/cluster-logging/elasticsearch-percentage-index-management.md

```diff
@@ -0,0 +1,156 @@
+
+This seeks to offer a mechanism by which one can denote the usage percentage (either of each node or of the total cluster) for indicating which indices should be deleted over.
+
+### User Stories
+
+#### As an user of EO, I want to be able to configure my delete policies based on a percent usage
+
+Currently we only offer a mechanism to delete indicies based on the age of the index.
+
+### Implementation Details
+
+#### Assumptions
+
+* This will be going out with our ES6 operator which means it will be a part of our ES 6.x cluster
+
+* It will be possible to configure phases based on both usage and time. We will trigger on whichever comes first.
+
+#### Security
+
+No additional security concerns need to be addressed by this.
+
+### Risks and Mitigations
+
+## Design Details
+
+We will add a field that is used by the delete phase. We will delete on either MinAge or Usage, whichever is evaluated as true first.
+
+### index_management_types.go
+
+We will add a usage field for our phases to use:
+
+```
+// +k8s:openapi-gen=true
+type IndexManagementDeletePhaseSpec struct {
+	// The minimum age of an index before it should be deleted (e.g. 10d)
+	//
+	MinAge TimeUnit `json:"minAge, optional"`
+
+  // The percent disk usage to wait for before deleting indices for this alias (e.g. 75)
+  //
+  Usage int32 `json:"usage, optional"`
+}
+
+```
+
+### Test Plan
+
+#### Unit Testing
+
+* We want to ensure that when Usage is specified we correctly update our delete cronjob scripts to use it.
+
+#### Integration and E2E tests
+
+* We want to ensure that when Usage is specified, see that we do not delete indices for that alias until disk utilization reaches the specified value.
+
+* We want to ensure that if both MinAge and Usage are specified, see that we will trigger and delete indices when one of the conditions are true.
+
+### Graduation Criteria
+
+#### GA
+
+* Gather feedback from users rather than just developers
+* End user documentation
+* Sufficient test coverage
+
+### Version Skew Strategy
+
+Given we are adding an optional field to an existing API there should be minimal concern for upgrading. This would also be happening as we are changing our release process from ART to CPaaS and therefore changing our operator versioning.
+
+## Implementation History
+
+| release|Description|
+|---|---|
+|5.1| **GA** - Initial release
+
+## Drawbacks
+
+Currently this is not possible in the OpenDistro Index Management plugin and causes us to deviate from what they offer. We would need to ensure that it is added by us prior to us moving to ES7 (and adopting their plugins).
+
+The naming convention "MinAge" when paired with "Usage" could give a false impression -- we may end up deleting indices before they reach the "MinAge" because they are the oldest indices and with "Usage" being reached. This may just need to be framed differently within the docs and field descriptions.
```

Comment:
We can frame this in the same context as the ES docs for the criteria by which they rollover an index: its when any of the 3 possible conditions is met

Reply:
that's what I convey in ##Design Details and more calling out here that we should make sure it is clear in the docs.